### ISSUE TITLE
The provider provider.azuread does not support resource type "azuread_application_password

### ISSUE THREAD

**Issue by Kumoservices** [terraform-providers/terraform-provider-azuread#186]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version
Terraform v0.12.17
+ provider.azuread v0.3.1
+ provider.azurerm v1.35.0
+ provider.random v2.2.1

### Affected Resource(s)

* `azuread_application_password`

### Terraform Configuration Files

```hcl
# Manages an Application within Azure Active Directory
resource "azuread_application" "start_stop_vm" {
  name                       = "${var.entity}-${var.keykats}-${var.environment}-acc-${var.app_code}_start_stop_vm"
  reply_urls                 = ["https://spn/${var.entity}-${var.keykats}-${var.environment}-acc-${var.app_code}_start_stop_vm"]
  available_to_other_tenants = false
  oauth2_allow_implicit_flow = true
}

# Manages a Service Principal associated with an Application (start stop vm) within Azure Active Directory.
resource "azuread_service_principal" "start_stop_vm" {
  application_id               = "${azuread_application.start_stop_vm.application_id}"

  tags = "${merge(local.mandatory_tags, local.extra_tags)}"
}

# Random a unique password for azure AD application
resource "random_string" "start_stop_vm" {
  length  = 32
  special = false
  upper   = true

  keepers = {
    application = "${azuread_application.start_stop_vm.id}"
  }
}

# Manages a Password associated with an Application within Azure Active Directory.
resource "azuread_application_password" "start_stop_vm" {
  application_id = "${azuread_application.start_stop_vm.id}"
  value          = "${random_string.start_stop_vm.result}"
  end_date       = "${var.end_date}"

  lifecycle {
   ignore_changes = ["value"]
 }
}

```

### Expected Behavior

Must works perfectly with azuread provider ?!?

### Actual Behavior

Error: Invalid resource type

  on start-stop-vm_solution.tf line 105, in resource "azuread_application_password" "start_stop_vm":
 105: resource "azuread_application_password" "start_stop_vm" {

The provider provider.azuread does not support resource type
"azuread_application_password".

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

- Running in azure West Europe
- Work Context : try to terraform Azure Start Stop VM solution using ARM template for solution deployement : I must code terraform Azure use "Run As Accounts" needs (Azure AD directory App / Certificate / Connection / Contributor role / Service principa)l in storage account to my ARM solution deployement works.

### References

* #0000

**Comment by dcfsc** [comment]
I believe `azuread_application_password` is a newer resource. Your azuread provider version is very old. The current version is ([see CHANGELOG](https://github.com/terraform-providers/terraform-provider-azuread/blob/master/CHANGELOG.md#070-november-15-2019)) 0.7.0 (November 15, 2019).

Review the CHANGELOG for new resources. I think you need to update your provider.

**Comment by katbyte** [comment]
Hi @Kumoservices,

@dcfsc is correct, `azuread_application_password ` was released in v0.4.0 (June 06, 2019), so upgrading to a newer provider version should fix this for you. As such i am going to close this but do reach out again if upgrading to the latest provider doesn't fix your issue!
